### ISSUE TITLE
unexpected platform type tensorflow_savedmodel for rul

### ISSUE THREAD

**Issue by GowthamKudupudi** [triton-inference-server/server#3026]
**Description**
I have built and installed the `server` and `tensorflow_backend`. Below is the output when I run the `tritonserver`
```
$ tritonserver --model-repository=`pwd`/enabled_models --log-warning 1 --log-error 1
I0618 07:08:13.449667 8390 metrics.cc:228] Collecting metrics for GPU 0: NVIDIA Tesla T4
2021-06-18 07:08:13.639625: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
I0618 07:08:13.682413 8390 tensorflow.cc:2165] TRITONBACKEND_Initialize: tensorflow
I0618 07:08:13.682445 8390 tensorflow.cc:2175] Triton TRITONBACKEND API version: 1.0
I0618 07:08:13.682451 8390 tensorflow.cc:2181] 'tensorflow' TRITONBACKEND API version: 1.0
I0618 07:08:13.682455 8390 tensorflow.cc:2205] backend configuration:
{}
I0618 07:08:13.957906 8390 pinned_memory_manager.cc:206] Pinned memory pool is created at '0x7f2476000000' with size 268435456
I0618 07:08:13.958445 8390 cuda_memory_manager.cc:103] CUDA memory pool is created on device 0 with size 67108864
E0618 07:08:13.962236 8390 model_repository_manager.cc:1916] Poll failed for model directory 'rul': unexpected platform type tensorflow_savedmodel for rul
I0618 07:08:13.962387 8390 server.cc:504]
+------------------+------+
| Repository Agent | Path |
+------------------+------+
+------------------+------+

I0618 07:08:13.962418 8390 server.cc:543]
+------------+-------------------------------------------------------+--------+
| Backend    | Path                                                  | Config |
+------------+-------------------------------------------------------+--------+
| tensorrt   | <built-in>                                            | {}     |
| tensorflow | /opt/tritonserver/backends/tensorflow1/libtriton_tens | {}     |
|            | orflow1.so                                            |        |
+------------+-------------------------------------------------------+--------+

I0618 07:08:13.962438 8390 server.cc:586]
+-------+---------+--------+
| Model | Version | Status |
+-------+---------+--------+
+-------+---------+--------+

I0618 07:08:13.962503 8390 tritonserver.cc:1659]
+----------------------------------+------------------------------------------+
| Option                           | Value                                    |
+----------------------------------+------------------------------------------+
| server_id                        | triton                                   |
| server_version                   | 2.10.0                                   |
| server_extensions                | classification sequence model_repository |
|                                  |  model_repository(unload_dependents) sch |
|                                  | edule_policy model_configuration system_ |
|                                  | shared_memory cuda_shared_memory binary_ |
|                                  | tensor_data statistics                   |
| model_repository_path[0]         | /home/gowtham/server/docs/examples/enabl |
|                                  | ed_models                                |
| model_control_mode               | MODE_NONE                                |
| strict_model_config              | 1                                        |
| pinned_memory_pool_byte_size     | 268435456                                |
| cuda_memory_pool_byte_size{0}    | 67108864                                 |
| min_supported_compute_capability | 6.0                                      |
| strict_readiness                 | 1                                        |
| exit_timeout                     | 30                                       |
+----------------------------------+------------------------------------------+

I0618 07:08:13.962551 8390 server.cc:234] Waiting for in-flight requests to complete.
I0618 07:08:13.962578 8390 server.cc:249] Timeout 30: Found 0 live models and 0 in-flight non-inference requests
error: creating server: Internal - failed to load all models
```
model load fails with `unexpected platform type tensorflow_savedmodel for rul`

**Triton Information**
r21.05

Are you using the Triton container or did you build it yourself?
I build it myself

**To Reproduce**
build the `server` with
`./build.py --no-container-build --target-platform ubuntu --cmake-dir=$(pwd)/build --build-dir=`pwd`/citritonbuild --enable-logging --enable-stats --enable-tracing --enable-metrics --enable-gpu-metrics --enable-gpu --filesystem=gcs --filesystem=azure_storage --filesystem=s3 --endpoint=http --endpoint=grpc --repo-tag=common:r21.05 --repo-tag=core:r21.05 --repo-tag=backend:r21.05 --repo-tag=thirdparty:r21.05 --backend=ensemble --backend=tensorrt --backend=python --backend=identity --backend=repeat --backend=square --repoagent=checksum
`
build the `tensorflow_backend` with
`cmake -DCMAKE_INSTALL_PREFIX:PATH=`pwd`/install -DTRITON_TENSORFLOW_VERSION=1 -DTRITON_TENSORFLOW_DOCKER_IMAGE="nvcr.io/nvidia/tensorflow:21.05-tf1-py3" -DTRITON_BACKEND_REPO_TAG=r21.05 -DTRITON_CORE_REPO_TAG=r21.05 -DTRITON_COMMON_REPO_TAG=r21.05 ..
`
copy install files to `/opt/tritonserver`
add bin and lib to path and run the tritonserver as mentioned in the description above

**Expected behavior**
hand built `tritonserver` should be able to recognize `tensorflow_savedmodel` platform just like the same model is loaded by the pre built container 21.05 without a problem.

**Comment by CoderHam** [comment]
You will need to specify the --backend=tensorflow1/2 flag so that the tensorflow1/2 backend is enabled. Currently you have just built the backend (libtriton_tensorflow1/2.so). 
For the no container build you will need to also specify these flags. 
```
    --library-paths=tensorflow1:/path/to/tensorflow1 \
    --library-paths=tensorflow2:/path/to/tensorflow2 \
```

**Comment by GuanLuo** [comment]
I assume that your model config doesn't specify `backend` to be tensorflow? If so, you should specify that or you should build server with `--backend=tensorflow1` / `--backend=tensorflow2`

**Comment by GowthamKudupudi** [comment]
@CoderHam and @GuanLuo, you can see in the above output of the `tritonserver` in the backend table, the tensorflow backend is loaded.

**Comment by GowthamKudupudi** [comment]
@GuanLuo I did add `backend: "tensorflow"` in the `config.pbtxt` of the model.

**Comment by GuanLuo** [comment]
We will track this issue and work on a fix. But for now, I think you would need to build Triton server with `--backend=tensorflow1` / `--backend=tensorflow2`, and you can either use the Tensorflow backend that comes with Triton or the one you built separately.

**Comment by dyastremsky** [comment]
Fixed with #3118. This should be working now with tensorflow as the backend. Your custom backend will perform the platform field checks.
